bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java

```diff
@@ -129,6 +131,22 @@ boolean complete(BookieSocketAddress host, final ByteBuf buffer) {
         }
 
         /**
+         * Fail the request with given result code <i>rc</i>.
+         *
+         * @param rc
+         *          result code to fail the request.
+         * @return true if we managed to fail the entry; otherwise return false if it already failed or completed.
```

Comment:
this change is to reading all replicas in parallel at the same time. the read completes whenever receiving a success response. A failure might come after a success response.